components/tidb_query_datatype/src/codec/table.rs

```diff
@@ -511,6 +510,33 @@ mod tests {
     const TABLE_ID: i64 = 1;
     const INDEX_ID: i64 = 1;
 
+    pub fn generate_index_data(
```

Comment:
It does not seem good that this function is repeated. I think we can just make it public without `#cfg[test]` gate.